Reconstruct _rs_graph and _rs_version_map when clone finished

chaoyli opened the pull request:
Upon finishing clone, tablet_meta has been changed.
_rs_graph and _rs_version_map should be reconstructed to be consistent with tablet_meta been changed.

Commit:
Reconstruct _rs_graph and _rs_version_map when clone finished

Upon finishing clone, tablet_meta has been changed.
_rs_graph and _rs_version_map should be reconstructed to be consistent with tablet_meta been changed.

kangpinghuang commented on be/src/olap/tablet.cpp:
```diff
@@ -206,6 +206,24 @@ OLAPStatus Tablet::revise_tablet_meta(const TabletMeta& tablet_meta,
 
+    for (auto& version : versions_to_delete) {
+        auto it = _rs_version_map.find(version);
+        _rs_version_map.erase(it);
```
_inc_rs_version_map need also erase it?

chaoyli replied:
I have missed to handle _inc_rs_version_map.

kangpinghuang commented on be/src/olap/tablet.cpp:
```diff
@@ -206,6 +206,24 @@ OLAPStatus Tablet::revise_tablet_meta(const TabletMeta& tablet_meta,
+    for (auto& rs_meta : rowsets_to_clone) {
+        Version version = { rs_meta->start_version(), rs_meta->end_version() };
+        RowsetSharedPtr rowset(new AlphaRowset(&_schema, _tablet_path, _data_dir, rs_meta));
+        _rs_version_map[version] = rowset;
```
I think rowset should be added to the map after init and you should return failure if init failed.

chaoyli replied:
OK

Commit:
Reconstruct _rs_graph and _rs_version_map when clone finished

Upon finishing clone, tablet_meta has been changed.
_rs_graph and _rs_version_map should be reconstructed to be consistent with tablet_meta been changed.
